pkg/kubelet/pluginmanager/plugin_manager_test.go

```diff
@@ -125,8 +126,12 @@ func TestPluginRegistration(t *testing.T) {
 	// Start the plugin manager
 	stopChan := make(chan struct{})
 	defer close(stopChan)
+
+	pluginServing := make(chan struct{})
+
 	go func() {
 		sourcesReady := config.NewSourcesReady(func(_ sets.String) bool { return true })
+		<-pluginServing
 		pluginManager.Run(sourcesReady, stopChan)
```

Comment:
how is this different than simply moving this goroutine after the p.Serve call?

Reply:
That's a good point! Updated as you said.